controller/volume_attachment_controller.go

```diff
@@ -482,39 +482,34 @@ func (vac *VolumeAttachmentController) shouldDoDetach(va *longhorn.VolumeAttachm
 	if len(currentAttachmentTickets) == 0 {
 		return true
 	}
-	if !hasUninterruptibleTicket(currentAttachmentTickets) && hasWorkloadTicket(attachmentTicketsOnOtherNodes) {
-		log.Info("Workload attachment ticket interrupted snapshot-controller/backup-controller attachment tickets")
-		return true
-	}
 
-	// Currently, the only ticket that is interruptible and frontend disabled is the rebuilding-controller ticket
-	// Offline replica rebuilding has the potential to fail and repeated attempts.
-	// Users can give up (interrupt) the replica rebuilding and attach the degraded volume to a node.
-	if hasInterruptibleAndFrontendDisabledTicket(currentAttachmentTickets) && hasWorkloadTicket(va.Spec.AttachmentTickets) {
-		log.Debugf("Workload attachment ticket interrupted rebuilding-controller attachment tickets")
+	if hasInterruptibleTicket(currentAttachmentTickets, vol) && hasWorkloadTicket(attachmentTicketsOnOtherNodes) {
+		log.Info("Workload attachment ticket interrupted snapshot-controller/backup-controller/rebuilding-controller attachment tickets")
 		return true
 	}
 
 	return false
 }
 
-func hasInterruptibleAndFrontendDisabledTicket(attachmentTickets map[string]*longhorn.AttachmentTicket) bool {
+func hasInterruptibleTicket(attachmentTickets map[string]*longhorn.AttachmentTicket, vol *longhorn.Volume) bool {
```

Comment:
Wait... The naming is wrong... When the list contains tickets for auto snapshot/backup/rebuilding, aka interruptible tickets, this function will return `false`... 
I prefer the previous logic: We should do detachment when there is no uninterruptible ticket on the current node but important tickets on other nodes. And with this logic, do we need to check `vol.Status.FrontendDisabled` still?

Reply:
Oh, yes. You're right. Don't need to check `vol.Status.FrontendDisabled` if we consider the tickets on other nodes.

`!hasUninterruptibleTicket` racked my brain...:(

Reply:
> !hasUninterruptibleTicket() racked my brain...:(

LOL. My brain is racked too but I couldn't find better name :D